gitlab.go

```diff
@@ -291,8 +294,7 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 
 	response := newResponse(resp)
 
-	err = CheckResponse(resp)
-	if err != nil {
+	if err := CheckResponse(resp); err != nil {
```

Comment:
Why change this? Seems just a matter of taste as the result is functionally 100% the same and technically is will add some (minor) costs for the additional shadow variable that this updated code will create.